src/redux/selectors/index.js

```diff
@@ -238,32 +244,85 @@ const sortedGroupedRecordsByType = (records, isDescending) => {
+    return sortedGroupedRecordsByType(items)
+      .filter((group) => group.type === selectedResourceType);
   },
 );
 
-const collectionItemsSelector = createSelector(
+const savedItemsSelector = createSelector(
   [resourcesSelector, activeCollectionResourceIdsSelector],
   (resources, saved) => values(resources)
     .filter(({ id }) => saved[id])
     .map(pickTimelineFields)
     .sort(sortByDate),
 );
 
-export const collectionRecordsGroupedByTypeSelector = createSelector(
-  [collectionItemsSelector, (_, ownProps) => ownProps],
-  (collectionItems, ownProps) => {
+// used by SubTypeAccordion in CatalogScreen and RecordType sorting in DetailsPanel
+export const savedRecordsGroupedByTypeSelector = createSelector(
+  [savedItemsSelector, activeCollectionSelector],
+  (savedItems, collection) => {
+    const { savedRecordsSortingState: sortingState } = collection;
+    const { RECORD_TYPE } = sortFields;
+    const isDescending = sortingState.sortDirections[RECORD_TYPE] === SORT_DESC;
+    return sortedGroupedRecordsByType(savedItems, isDescending);
+  },
+);
+
+const sortedRecordsByRecordDate = (records, isDescending) => {
+  const sortedRecords = isDescending ? [...records].reverse() : records;
+  const typeMap = sortedRecords
+    .reduce((acc, record) => {
+      const { timelineDate, subType, type } = record;
+      const formattedDay = formatDate(timelineDate);
+      return produce(acc, (draft) => {
+        // eslint-disable-next-line no-param-reassign
+        draft[formattedDay] = draft[formattedDay] ?? {};
+        // eslint-disable-next-line no-param-reassign
+        draft[formattedDay][type] = draft[formattedDay][type] ?? {};
+        // eslint-disable-next-line no-param-reassign
+        draft[formattedDay][type][subType] = draft[formattedDay][type][subType] ?? [];
+        // eslint-disable-next-line no-param-reassign
+        draft[formattedDay][type][subType].push(record.id);
+      });
+    }, {});
+
+  return Object
+    .entries(typeMap)
+    .reduce((acc1, [date, typeValues]) => {
+      const typeData = Object
+        .entries(typeValues)
+        .reduce((acc2, [type, subTypeValues]) => {
+          const subTypeData = Object
+            .entries(subTypeValues)
+            .reduce((acc3, [subType, recordIds]) => acc3.concat({
+              subType,
+              recordIds,
+            }), []);
+          return acc2.concat({
+            type,
+            label: SINGULAR_RESOURCE_TYPES[type],
+            subTypes: subTypeData,
+          });
+        }, []);
+      return acc1.concat({
+        date,
+        types: typeData,
+      });
+    }, []);
+};
+
+export const savedRecordsByRecordDateSelector = createSelector(
+  [savedItemsSelector, (_, ownProps) => ownProps],
+  (items, ownProps) => {
     const { isDescending } = ownProps;
-    return sortedGroupedRecordsByType(collectionItems, isDescending);
+    return sortedRecordsByRecordDate(items, isDescending);
```

Comment:
if `isDescending` is (eventually?) in redux state, it could be derived via a selector composed from the active collection selector